libs/sm/drivers/LVHDSR.py

```diff
@@ -641,6 +641,7 @@ def detach(self, uuid):
         # However, we should still delete lock files on slaves as it is the
         # only place to do so.
         self._cleanup(self.isMaster)
+        cleanup.stop_gc_service(sr_uuid)
```

Comment:
This should replace the `cleanup.abort()` at line 589, or alternatively make `cleanup.abort()` stop the service.